packages/ui-components/src/components/text-field/text-field.tsx

```diff
@@ -0,0 +1,78 @@
+import { Component, Event, EventEmitter, h, Host, Prop, State, Watch } from '@stencil/core';
+import { EInputFieldType } from './text-field.types';
+
+@Component({
+	tag: 'kv-text-field',
+	styleUrl: 'text-field.scss',
+	shadow: true
+})
+export class KvTextField {
+	/** (optional) Text field tyoe */
+	@Prop({ reflect: true }) type!: EInputFieldType;
+	/** (optional) Text field label */
+	@Prop({ reflect: true }) label: string;
+	/** (optional) Text field input name */
+	@Prop({ reflect: true }) inputName: string;
+	/** (optional) Text field place holder */
+	@Prop({ reflect: true }) placeholder: string;
+	/** (optional) Text field disabled */
+	@Prop({ reflect: true }) disabled = false;
+	/** (optional) Text field required */
+	@Prop({ reflect: true }) required = false;
+	/** (optional) Text field loading state */
+	@Prop({ reflect: true }) loading = false;
+	/** (optional) Text field help text */
+	@Prop({ reflect: true }) helpText: string;
+	/** (optional) Text field slim style */
+	@Prop({ reflect: true }) slim = false;
+
+	/** Text field value */
+	@Prop({ reflect: true, mutable: true }) value: string;
+
+	/** Text field value state */
+	@State() _value: string;
+
+	/** Emitted when text field's value changes */
+	@Event() textChange: EventEmitter<InputEvent>;
+
+	/** Emitted when text field's value changes */
+	@Event() textFieldBlur: EventEmitter<FocusEvent>;
+
+	/** Watch `value` property for changes and update `_value` accordingly */
+	@Watch('value')
+	valueChangeHandler(newValue: string) {
+		this._value = newValue;
+	}
+
+	private handleChange(event) {
```

Comment:
change this to an arrow function, this way you can reference it directly in JSX ;)